adserver/models.py

```diff
@@ -1421,15 +1421,28 @@ def __copy__(self):
         # Get a fresh reference so that "self" doesn't become the new copy
         ad = Advertisement.objects.get(pk=self.pk)
 
+        new_name = ad.name
+        new_slug = ad.slug
+
+        # Fix up names of ads that have been copied multiple times
+        while new_name.endswith(" Copy"):
+            new_name = new_name[:-5]
+        new_slug = re.sub("-copy\d*$", "", new_slug)
+
         # Get a slug that doesn't already exist
-        new_slug = ad.slug + "-copy"
+        # This tries -copy, -copy1, -copy2, etc.
+        new_slug += "-copy"
+        digit = 0
```

Comment:
I wonder about using something that will change over time, like a timestamp or similar, instead of `copy`? I could see something like `datadog-us-data-science-rev1-202306` or something? Could perhaps bringing additional complexity for minimal value though..

Reply:
Not 100% sure, but seems more explicit than just `copy2`, to show when it was created? 🤷 It might make it more cluttered, but `Copy Copy Copy` shows users don't really care much about the name, unless they're using the slug in reports.

Reply:
Ok, I added a change to use the date. It prunes the date when copying.